### ISSUE TITLE
Editor/Publisher groups per content type

### ISSUE THREAD

**Issue by aaronbushnell** [engram-design/Workflow#16]
First off, _awesome_ plugin! This is extremely well done.

I had a question about how Workflow should (or could) be setup to handle a particular user group/content type scenario. Here's the roles and the content types these groups have access to.

## PR Team (Editors)
- News
- Blog Posts
- Calendar Events

## PR Team (Publishers)
- News
- Blog Posts
- Calendar Events

## News Team (Editors)
- News
- Blog Posts

## News Team (Publishers)
- News
- Blog Posts

I have multiple Editor and Publisher groups, but Workflow expects a single Editor group and Publisher group. My groups also vary on content type and even have content type overlap. Does Workflow have a way to manage this type of setup? Or will it at some point?

**Comment by raptureaid** [comment]
You should be able to accomplish that by creating parent/child groups for both editors and publishers. All of your permissions (create, edit, etc.) would be set on the child group(s), the parent group controls access to Workflow features. In Workflow settings, point Editor/Publisher user groups to each parent. Then, assign users to both the parent and the team specific child group(s).

Editor Parent (no permissions)
|---- Editor PR Team (team specific permissions)
|---- Editor News Team (team specific permissions)

Publisher Parent (no permissions)
|---- Publisher PR Team (team specific permissions)
|---- Publisher News Team (team specific permissions)

**Comment by aaronbushnell** [comment]
**Great** idea! Didn't even consider this approach. So, just to confirm, you're saying these are the user groups I'd end up with:

```
Editor
Publisher
PR Team
News Team
```

And the steps are as follows:

1. Create two primary user groups with no permissions ("Editor" and "Publisher")
2. Set up Workflow to use these as the primary "Editor" and "Publisher" groups
3. Create any additional user groups as needed ("PR Team", "News Team", etc)
4. When assigning a new user to a group use _both_ types groups. The primary group ("Editor" or "Publisher") and the secondary group ("PR Team", "News Team", etc).

Am I understanding that correctly?

**Comment by raptureaid** [comment]
Yes, you would assign users to two groups. The trick is in where the permissions get assigned.

The primary group (editor/publisher) should not define any permissions - it's essentially a placeholder for the Workflow plugin.

The secondary groups are where all the permissions get defined - can create, can edit, can publish, etc. You need at least one secondary group for editors (no publish permission) and publishers. You can have any number of secondary groups (editor or publisher) as needed.

Given your example, you would create four secondary groups:
* PR team editor
* PR team publisher
* News editor
* News publisher

Then configure the permissions for the groups as required. For example, PR team editor would get access to create and edit news, blogs and events; PR team publisher would set permissions to publish news, blogs and events. Repeat for News team.

Finally, assign users the both a primary and secondary group:
* Editor + PR team editor
* Editor + News team editor
* Publisher + PR team publisher
* Publisher + News team publisher

**Comment by Emkaytoo** [comment]
I suspect that a user could belong to multiple workflows by being a member of multiple groups, yes? So a user could be a Publisher for PR team but be only an Editor for News?

**Comment by raptureaid** [comment]
That level of granularity does not seem possible at the moment. Using multiple groups comes close, but is still only a workaround to limit editor/publisher controls to specific sections of the CMS. While you can have many subgroups of each, editors and publishers are still unique and can't crossover. Adding a user as an editor in one area and a publisher in another would cause a permission mismatch.

**Comment by aaronbushnell** [comment]
Great info, @raptureaid. Perhaps one day Workflow will allow the type of functionality @Emkaytoo is asking for. But for now, I believe this covers my particular use-case pretty nicely.

Appreciate the stellar work on this!

**Comment by aaronbushnell** [comment]
@raptureaid Had an additional question here. If you have multiple groups to handle permissions...
```
Editor Master (no permissions)
|---- Editor PR Team (team specific permissions)
|---- Editor News Team (team specific permissions)

Publisher Master (no permissions)
|---- Publisher PR Team (team specific permissions)
|---- Publisher News Team (team specific permissions)
```
...and a "News Team (Editor)" requests approval of their new "News" entry won't the email notification go to everyone in the "Publisher" group? Wouldn't these go to users who shouldn't be administrating the "News" channel?
